### PR TITLE
ComboBox: Option to use ariaLabel prop as preview text

### PR DESCRIPTION
**Pull request by kysedate**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [x] Include a change request file using `$ npm run change`

#### Description of changes

There are scenarios where the text passed into a ComboBox option isn't plain text, but rather text with an embedded style. In this case, we want to render a plain text version of the string on the aria-label of the option, in the ComboBox input when selected, and used for that option's autocomplete string. This prop allows for the ComboBox to handle this case by using the ariaLabel prop instead of the text prop when necessary.

### CODE REVIEW

File: packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx [hunk 1]
```diff
@@ -583,18 +583,21 @@ export class ComboBox extends BaseComponent<IComboBoxProps, IComboBoxState> {
     if (this.props.autoComplete === 'on') {
 
       // If autoComplete is on, attempt to find a match where the text of an option starts with the updated value
-      const items = currentOptions.map((item, index) => { return { ...item, index }; }).filter((option) => option.itemType !== SelectableOptionMenuItemType.Header && option.itemType !== SelectableOptionMenuItemType.Divider).filter((option) => option.text.toLocaleLowerCase().indexOf(updatedValue) === 0);
+      const items = currentOptions.map((item, index) => { return { ...item, index }; }).filter((option) => option.itemType !== SelectableOptionMenuItemType.Header && option.itemType !== SelectableOptionMenuItemType.Divider).filter((option) => this._getPreviewText(option).toLocaleLowerCase().indexOf(updatedValue) === 0);
```
Comment (jspurlin) [inline packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx]: This line is really long, consider breaking it up across multiple line

File: packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx [hunk 2]
```diff
@@ -583,18 +583,21 @@ export class ComboBox extends BaseComponent<IComboBoxProps, IComboBoxState> {
...
+        newSuggestedDisplayValue = text.toLocaleLowerCase() !== updatedValue ? text : '';
 
         // remember the index of the match we found
         newCurrentPendingValueValidIndex = items[0].index;
       }
     } else {
 
       // If autoComplete is off, attempt to find a match only when the value is exactly equal to the text of an option
-      const items = currentOptions.map((item, index) => { return { ...item, index }; }).filter((option) => option.itemType !== SelectableOptionMenuItemType.Header && option.itemType !== SelectableOptionMenuItemType.Divider).filter((option) => option.text.toLocaleLowerCase() === updatedValue);
+      const items = currentOptions.map((item, index) => { return { ...item, index }; }).filter((option) => option.itemType !== SelectableOptionMenuItemType.Header && option.itemType !== SelectableOptionMenuItemType.Divider).filter((option) => this._getPreviewText(option).toLocaleLowerCase() === updatedValue);
```
Comment (jspurlin) [inline packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx]: same comment here

File: packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx [hunk 3]
```diff
@@ -1040,7 +1043,7 @@ export class ComboBox extends BaseComponent<IComboBoxProps, IComboBoxState> {
           onMouseLeave={ this._onOptionMouseLeave }
           role='option'
           aria-selected={ isSelected ? 'true' : 'false' }
-          ariaLabel={ item.text }
+          ariaLabel={ this._getPreviewText(item) }
```
Comment (jspurlin) [inline packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx]: What about the multiSelect case?

File: packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.types.ts [hunk 4]
```diff
@@ -26,6 +26,12 @@ export interface IComboBoxOption extends ISelectableOption {
    * the prop comboBoxOptionStyles
    */
   styles?: Partial<IComboBoxOptionStyles>;
+
+  /**
+   * In scenarios where embedded data is used at the text prop, we will use the ariaLabel prop to set the aria-label and preview text. Default to false
```
Comment (jspurlin) [inline packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.types.ts]: NOT: wrap this to multiple lines as well

File: packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx (hunk 3 above)
Comment (kysedate) [reply packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx]: I'm not sure what you mean..

File: packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx [hunk 5]
```diff
@@ -1764,4 +1767,8 @@ export class ComboBox extends BaseComponent<IComboBoxProps, IComboBoxState> {
 
     return retKeys;
   }
+
+  private _getPreviewText(item: IComboBoxOption): string {
```
Comment (ddlbrena) [inline packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx]: this seems like a good place to add a comment explaining why are we doing this

File: packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx (hunk 3 above)
Comment (jspurlin) [reply packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx]: the checkbox below seems to be missing the ariaLabel... we should fix that

**Pull request merged**